SensoBackend/Application/Modules/Alerts/CreateAlert.cs

```diff
@@ -0,0 +1,46 @@
+using FluentValidation;
+using JetBrains.Annotations;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using SensoBackend.Application.Modules.Alerts.Contracts;
+using SensoBackend.Application.Modules.Alerts.Utils;
+using SensoBackend.Domain.Entities;
+using SensoBackend.Domain.Exceptions;
+using SensoBackend.Infrastructure.Data;
+
+namespace SensoBackend.Application.Modules.Alerts;
+
+public sealed record CreateAlertRequest : IRequest
+{
+    public required int AccountId { get; init; }
+    public required CreateAlertDto Dto { get; init; }
+}
+
+[UsedImplicitly]
+public sealed class CreateAlertHandler(AppDbContext context, TimeProvider timeProvider)
+    : IRequestHandler<CreateAlertRequest>
+{
+    public async Task Handle(CreateAlertRequest request, CancellationToken ct)
+    {
+        var seniorProfile =
+            await context
+                .Profiles
+                .FirstOrDefaultAsync(
+                    p => p.AccountId == request.AccountId && p.SeniorId == request.AccountId,
+                    ct
+                ) ?? throw new SeniorNotFoundException(request.AccountId);
+
+        var alertType = GetAlertType.FromName(request.Dto.Type);
+
+        var alert = new Alert
+        {
+            Id = default,
+            SeniorId = seniorProfile.SeniorId,
+            Type = alertType,
+            FiredAt = timeProvider.GetUtcNow(),
+        };
```

Comment:
It would be great to create a separate request and request handler for actual alert creation, which would take an `Alert` object as a parameter and add it to a database. We could then call that request using mediator from this one. It seems like unnecessary abstraction, but if we did that then we can reuse that request whenever we add notifications triggered by the system (for missed medication etc.). And whenever we add notifications to alert creation, we could also put that there instead of copying it over all of the actions.

That's how I see it:
```
 ┌───────────────────┐
 │POST /api/v1/alerts│
 └─────────┬─────────┘
           │
           │ CreateSosAlertRequest
           │
┌──────────▼──────────┐
│CreateSosAlertHandler│
└──────────┬──────────┘
           │
           │ CreateAlertRequest
           │
 ┌─────────▼────────┐ CreateAlertRequest ┌──────────┐
 │CreateAlertHandler◄────────────────────┤CronModule│
 └──────────────────┘                    └──────────┘
  - sends the notification                - sends a request on
  - add the alert to the DB                 missed medication etc.
```

Reply:
We could make it `DispatchAlertHandler` instead to have less confusing naming.